src/1.generic-constraints/index.ts

```diff
@@ -4,6 +4,24 @@
 // You can remove the readonly modifier from the mapped type by using -readonly
 
 // ------- start coding here -------
+//Definiskan fungsi generik `mapObjectProperties`
+function mapObjectProperties<
+  S extends object,
+  M extends Record<keyof M, keyof S>
+>(
+  source: S,
+  mapping: M
+): {
+  [K in keyof M]: S[M[K]];
```

Comment:
```suggestion
  [K in keyof M]: S[M[K]];
```
Kode ini silahkan dibuat menggunakan sebuah type sendiri yang nantinya akan dipanggil dalam function atau tidak dibuat secara literal.
Tips: pada type yang sudah dibuat, silahkan menggunakan `-readonly` untuk menghilangkan readonly